src/mlpack/methods/reinforcement_learning/environment/bit_flipping.hpp

```diff
@@ -0,0 +1,256 @@
+   */
+  double Sample(const State& state,
+                const Action& action,
+                State& nextState,
+                const State& transitionGoal)
+  {
+    // Update the number of steps performed.
+    stepsPerformed++;
+
+    // // Modify state according to action
+    nextState.Data()= state.Data();
+    nextState.Data()(action.action) = 1 - state.Data()(action.action);
+
+    // Check if the episode has terminated.
+    bool done = IsTerminal(nextState);
+
+    // Do not reward agent if it failed.
+    if (done && arma::approx_equal(state.Data(), goal, "absdiff", 1e-5))
+      return 1.0;
+
+    // Reward agent if it reaches the goal of transition and is not done
+    if (!done && arma::approx_equal(nextState.Data(), transitionGoal.Data(),
+         "absdiff", 1e-5))
+    {
+      return 1.0;
+    }
+
+    return 0.0;
+  }
+
+  /**
+   * Get reward based on current state and current
+   * action.
+   *
+   * @param state The current state.
+   * @param action The current action.
+   * @param transitionGoal The transition goal.
+   * @return reward, it's always 1.0.
+   */
+  double Sample(const State& state, const Action& action,
+                 const State& transitionGoal)
+  {
+    State nextState;
+    return Sample(state, action, nextState, transitionGoal);
+  }
+
+  /**
+   * Initial state representation is randomly generated binary vector
+   *
+   * @return Initial state for each episode.
+   */
+  State InitialSample()
+  {
+    stepsPerformed = 0;
+    initialState = arma::randi<arma::vec>(length, arma::distr_param(0, 1));
+    return State(initialState);
+  }
+
+  /**
+   * Get reward for particular goal
+   *
+   * @param nextState The next state.
+   * @param transitionGoal Transition's goal.
+   * @return Initial state for each episode.
+   */
+  double GetHERReward(const State& nextState,
+                    const State& transitionGoal)
+  {
+    // Reward agent if it reaches the goal of transition and is not done
+    if (arma::approx_equal(nextState.Data(), transitionGoal.Data(), 
+        "absdiff", 1e-5))
+    {
+      return 1.0;
+    }
+
+    return 0.0;
+  }
+
+  /**
+   * This function checks if the cart has reached the terminal state.
+   *
+   * @param state The desired state.
+   * @return true if state is a terminal state, otherwise false.
+   */
+  bool IsTerminal(const State& state) const
+  { 
+    if (maxSteps != 0 && stepsPerformed >= maxSteps)
+    {
+      Log::Info << "Episode terminated due to the maximum number of steps"
+          "being taken.";
+      return true;
+    }
+    else if (arma::approx_equal(state.Data(), goal, "absdiff", 1e-5))
+    {
+      Log::Info << "Episode terminated as agent has reached desired goal.";
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Initial goal representation for thr environment
+   *
+   * @return Initial goal for each episode.
+   */
+  State GoalSample()
+  {
+    goal = arma::ones(length);
```

Comment:
~length~ bitWidth?

Reply:
Sure that seems more appropriate. I will make that change.